### PR TITLE
add max program size to settings

### PR DESCRIPTION
**Pull request by yuval-k**
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/2805

### CODE REVIEW

File: pkg/utils/regex/regex.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+	settings := settingsutil.MaybeFromContext(ctx)
+	var programsize *uint32
+	if settings != nil {
+		if max_size := settings.GetGloo().GetRegexMaxProgramSize(); max_size != nil {
+			programsize = &max_size.Value
+		}
+	}
+
+	return NewRegexWithProgramSize(regex, programsize)
```
Comment (kdorosh) [inline pkg/utils/regex/regex.go:22]: we could DRY this up by just calling `NewRegexFromSettings()`

Comment (kdorosh) [reply pkg/utils/regex/regex.go:22]: also I don't think the nil handling here works (do we want max programsize of zero??)

File: projects/gloo/api/v1/settings.proto [hunk 2]
```diff
@@ -386,6 +386,9 @@ message GlooOptions {
     // set to false if you wish to reset the envoy configuration to a clean slate with no routes.
     // If not specified, defaults to `false`.
     google.protobuf.BoolValue disable_proxy_garbage_collection = 9;
+
+    // Set this option to specify the default max program size for regexs.
```
Comment (kdorosh) [inline projects/gloo/api/v1/settings.proto:390]: the plural of regex is regexes

Comment (kdorosh) [reply projects/gloo/api/v1/settings.proto:390]: what is the default if this is omitted? (100 iirc)

Comment (yuval-k) [reply projects/gloo/api/v1/settings.proto:390]: good point, will update

**Pull request merged**